RAMP/ramp.py

```diff
@@ -68,34 +70,29 @@ def unpack(bundle):
 
 @ramp.command()
 @click.argument('module')
-@click.option('--output', '-o', default='module.zip', help='output file name')
-@click.option('--verbose', '-v', is_flag=True, help='verbose mode: print the resulting metadata')
 @click.option('--manifest', '-m', type=click.File('rb'), help='generate package from manifest')
-@click.option('--display-name', '-d', 'display_name', default=module_metadata.DISPLAY_NAME, help='name for display purposes')
-@click.option('--module-name', '-n', 'module_name', help='module name')
-@click.option('--author', '-a', default=module_metadata.AUTHOR, help='module author')
-@click.option('--email', '-e', default=module_metadata.EMAIL, help='author\'s email')
-@click.option('--architecture', '-A', default=module_metadata.ARCHITECTURE, help='module compiled on i386/x86_64 arch')
-@click.option('--description', '-D', default=module_metadata.DESCRIPTION, help='short description')
-@click.option('--homepage', '-h', default=module_metadata.HOMEPAGE, help='module homepage')
-@click.option('--license', '-l', default=module_metadata.LICENSE, help='license')
-@click.option('--cmdargs', '-c', default=module_metadata.COMMAND_LINE_ARGS, help='module command line arguments')
-@click.option('--redis-min-version', '-r', 'redis_min_version', default=module_metadata.MIN_REDIS_VERSION, help='redis minimum version')
-@click.option('--redis-pack-min-version', '-R', 'redis_pack_min_version', default=module_metadata.MIN_REDIS_PACK_VERSION, help='redis pack minimum version')
-@click.option('--config-command', '-cc', default=module_metadata.CONFIG_COMMAND, help='command used to configure module args at runtime')
-@click.option('--os', '-O', default=module_metadata.OS, help='build target OS (Darwin/Linux)')
+@click.option('--display-name', '-d', default=None, help='name for display purposes')
```

Comment:
Why did you removed all default values?

Reply:
First, there are two kind of arguments there: ones that correspond to RAMP fields and ones that don't. The latter kind (listed at the end) have default values. For the "field-args", we have to know whether they are passed by the user, as we want them to override default and manifest values. By giving them a default value of None we can detect that. Note that default values are provided by create_default_metadata(), and used to initialize the metadata structure before overriding it with manifest and command-line values.